Sources/FoundationEssentials/FileManager/FileOperations.swift

```diff
@@ -960,6 +960,45 @@ enum _FileOperations {
     private static func _copyDirectoryMetadata(srcFD: CInt, srcPath: @autoclosure () -> String, dstFD: CInt, dstPath: @autoclosure () -> String, delegate: some LinkOrCopyDelegate) throws {
         #if !os(WASI) && !os(Android)
         // Copy extended attributes
+        #if os(FreeBSD)
+        // FreeBSD uses the `extattr_*` calls for setting extended attributes. Unlike like, the namespace for the extattrs are not determined by prefix of the attribute
+        for namespace in [EXTATTR_NAMESPACE_SYSTEM, EXTATTR_NAMESPACE_USER] {
+            // if we don't have permission to list attributes in system namespace, this returns -1 and skips it
+            var size = extattr_list_fd(srcFD, namespace, nil, 0)
+            if size > 0 {
+                try withUnsafeTemporaryAllocation(of: CChar.self, capacity: size + 1) { keyList in
```

Comment:
It took me a moment to discern from reading the code, but it seems like the extra `+ 1` byte is to have room to add a null terminator at the end of the last name. Could you add a comment explaining that here since at first I expected this to just allocate `size` bytes?